icelake/src/io/writer_builder.rs

```diff
@@ -142,7 +143,7 @@ impl<L: FileAppenderLayer<DefaultFileAppender>> WriterBuilder<L> {
     pub async fn build_equality_delta_writer(
         self,
         unique_column_ids: Vec<usize>,
-    ) -> Result<EqualityDeltaWriter<impl FileAppenderFactory>> {
+    ) -> Result<EqualityDeltaWriter<FileAppenderBuilder<L>>> {
```

Comment:
Is it possible to make the return type:
```
Box<dyn FileAppdnerFactory<R=Box<dyn FileAppender>>
```

I don't want to leak `FileAppenderBuilder` outside.

Reply:
After remove `FileAppenderFactory`, we can just expose the `FileAppenderLayer`.